lib/substitute.js

```diff
@@ -0,0 +1,36 @@
+
+/**
+ * Expose `substitute`
+ */
+
+module.exports = substitute;
+
+/**
+ * Type.
+ */
+
+var type = Object.prototype.toString;
+
+/**
+ * Substitute `:prop` with the given `obj` in `str`
+ *
+ * @param {String} str
+ * @param {Object or Array} obj
+ * @param {RegExp} expr
+ * @return {String}
+ * @api public
+ */
+
+function substitute(str, obj, expr){
```

Comment:
Is the third parameter in this function used in metalsmith-permalinks? If not, it might be better to just remove that functionality entirely.